File: src/main/com/dune/ide/files/DuneFileType.java
package com.dune.ide.files;

import com.dune.DuneBundle;
import com.dune.icons.DuneIcons;
import com.dune.DuneLanguage;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class DuneFileType extends LanguageFileType {

    // instance
    public static final DuneFileType INSTANCE = new DuneFileType();
    private DuneFileType() {
        super(DuneLanguage.INSTANCE);
    }

    // implementation

    @Override @NotNull public String getName() {
        return "DUNE";
    }

    @Override @NotNull public String getDescription() {
        return DuneBundle.message("filetype.dune.description");
    }

    @Override @NotNull public String getDefaultExtension() {
        return "";
    }

    @Override public Icon getIcon() {
        return DuneIcons.FileTypes.DUNE_FILE;
    }

    @Override public @Nls @NotNull String getDisplayName() {
        return getDescription();
    }
}

File: src/main/kotlin/com/dune/ide/files/DuneFileType.kt
package com.dune.ide.files

import com.dune.DuneBundle
import com.dune.icons.DuneIcons
import com.dune.language.DuneLanguage
import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object DuneFileType : LanguageFileType(DuneLanguage) {
    // LanguageFileType implementation
    override fun getName(): String  = "Dune File"
    override fun getDescription(): String = DuneBundle.message("filetype.dune.description")
    override fun getDefaultExtension(): String = ""
    override fun getIcon(): Icon = DuneIcons.FileTypes.DUNE_FILE
}